### PR TITLE
Feat: new custom role resource

### PR DESCRIPTION
**Pull request by TomerHeber**
Part of #498 effort.

### Solution

1. added custom role assignment for "env0_team_project_assignment".
2. minor fixes/refactoring.
3. added acceptance tests.
4. added integration tests.

### CODE REVIEW

File: client/team_project_assignment.go [hunk 1]
```diff
@@ -18,32 +18,19 @@ func IsBuiltinProjectRole(role string) bool {
...
-	ProjectId   string      `json:"projectId"`
-	ProjectRole ProjectRole `json:"projectRole" tfschema:"role"`
+	Id          string `json:"id"`
+	TeamId      string `json:"teamId"`
+	ProjectId   string `json:"projectId"`
+	ProjectRole string `json:"projectRole" tfschema:"-"`
 }
 
 func (client *ApiClient) TeamProjectAssignmentCreateOrUpdate(payload TeamProjectAssignmentPayload) (TeamProjectAssignment, error) {
-	if payload.ProjectId == "" {
```
Comment (TomerHeber) [inline client/team_project_assignment.go:34]: These validations aren't needed.
Validated by the schema.

File: client/team_project_assignment_test.go [hunk 2]
```diff
@@ -48,41 +46,6 @@ var _ = Describe("TeamProjectAssignment", func() {
 				Expect(teamProjectAssignment).To(Equal(mockTeamProjectAssignment))
 			})
 		})
-
-		Describe("Failure", func() {
-
-			It("Should fail if assignment has no project_id", func() {
```
Comment (TomerHeber) [inline client/team_project_assignment_test.go:54]: removed the tests for these validations.

File: env0/resource_team_project_assignment.go [hunk 3]
```diff
@@ -57,20 +58,25 @@ func resourceTeamProjectAssignmentRead(ctx context.Context, d *schema.ResourceDa
 		return diag.Errorf("could not get TeamProjectAssignment: %v", err)
 	}
 
-	found := false
```
Comment (TomerHeber) [inline env0/resource_team_project_assignment.go:60]: I don't see a reason why this is needed.
Specifically, why d.IsNewResource is needed.

**Review by Yossi-kerner — APPROVED** [review]
LGTM

**Review by GiliFaroEnv0 — APPROVED** [review]
outstanding job! as always 😃

**Pull request merged**